contrib/platform/src/com/sun/jna/platform/mac/CoreFoundationUtil.java

```diff
@@ -0,0 +1,142 @@
+ *
+ * You may obtain a copy of the Apache License at:
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.mac;
+
+import static com.sun.jna.platform.mac.CoreFoundation.kCFStringEncodingUTF8;
+
+import java.util.Collection;
+
+import com.sun.jna.Memory;
+import com.sun.jna.platform.mac.CoreFoundation.CFAllocatorRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFNumberType;
+import com.sun.jna.platform.mac.CoreFoundation.CFTypeRef;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.LongByReference;
+
+/**
+ * Provides utilities for Core Foundations
+ */
+public class CoreFoundationUtil {
+
+    private static final CoreFoundation CF = CoreFoundation.INSTANCE;
+
+    public static final CFAllocatorRef ALLOCATOR = CF.CFAllocatorGetDefault();
+
+    private CoreFoundationUtil() {
+    }
+
+    /**
+     * Convert a reference to a Core Foundations LongLong into its {@code long}
+     *
+     * @param theLong
+     *            The pointer to a 64-bit integer
+     * @return The corresponding {@code long}
+     */
+    public static long cfPointerToLong(CFTypeRef theLong) {
+        LongByReference lbr = new LongByReference();
+        CF.CFNumberGetValue(theLong, CFNumberType.kCFNumberLongLongType.ordinal(), lbr);
+        return lbr.getValue();
+    }
+
+    /**
+     * Convert a reference to a Core Foundations Int into its {@code int}
+     *
+     * @param p
+     *            The pointer to an integer
+     * @return The corresponding {@code int}
+     */
+    public static int cfPointerToInt(CFTypeRef p) {
+        IntByReference ibr = new IntByReference();
+        CF.CFNumberGetValue(p, CFNumberType.kCFNumberIntType.ordinal(), ibr);
+        return ibr.getValue();
+    }
+
+    /**
+     * Convert a reference to a Core Foundations Boolean into its {@code boolean}
+     *
+     * @param cfBoolean
+     *            The pointer to a boolean
+     * @return The corresponding {@code boolean}
+     */
+    public static boolean cfPointerToBoolean(CFTypeRef cfBoolean) {
+        return CF.CFBooleanGetValue(cfBoolean);
+    }
+
+    /**
+     * Convert a reference to a Core Foundations String into its
+     * {@link java.lang.String}
+     *
+     * @param cfTypeRef
+     *            The pointer to a CFString
+     * @return The corresponding {@link java.lang.String}
+     */
+    public static String cfPointerToString(CFTypeRef cfTypeRef) {
+        if (cfTypeRef == null) {
+            return "null";
+        }
+        long length = CF.CFStringGetLength(cfTypeRef);
+        long maxSize = CF.CFStringGetMaximumSizeForEncoding(length, kCFStringEncodingUTF8);
+        if (maxSize == CoreFoundation.kCFNotFound) {
+            maxSize = 4 * (length + 1);
+        }
+        Memory buf = new Memory(maxSize);
+        CF.CFStringGetCString(cfTypeRef, buf, maxSize, kCFStringEncodingUTF8);
```

Comment:
I get the feeling, that I'm looking at the wrong documentation (https://developer.apple.com/documentation/corefoundation/1542721-cfstringgetcstring?language=objc):

> The string encoding to which the character contents of theString should be converted. The encoding must specify an 8-bit encoding.

UTF-8 is multi-byte....

Reply:
This only needs to map to a C String, which is an array of `char`.  If a string in C is Unicode, wouldn't it be stored this way?

Reply:
I've updated the unit test in `CoreFoundationTest` to use a Unicode string to show this works as intended.

Reply:
As long as it works I'm ok with it, the documentation is at least misleading.